server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -141,6 +144,27 @@ public static void download(TranslogTransferManager translogTransferManager, Pat
         }
     }
 
+    private static void deleteTranslogFilesNotUploaded(Path location, long uploadedGeneration) throws IOException {
+        // Delete translog files with generation > translogMetadata.getGeneration()
+        List<Long> generationsNotUploaded = Arrays.stream(FileSystemUtils.files(location))
+            .map(filePath -> filePath.getFileName().toString())
+            .filter(filename -> filename.endsWith(TRANSLOG_FILE_SUFFIX))
+            .map(filename -> {
+                long generation = Long.parseLong(
+                    filename.substring(TRANSLOG_FILE_PREFIX.length(), filename.length() - TRANSLOG_FILE_SUFFIX.length())
+                );
+                return generation > uploadedGeneration ? generation : -1;
+            })
+            .filter(generation -> generation > -1)
+            .collect(Collectors.toList());
+        for (Long generation : generationsNotUploaded) {
+            Path checkpointFileName = location.resolve(Translog.getCommitCheckpointFileName(generation));
+            Path translogFileName = location.resolve(Translog.getFilename(generation));
+            Files.deleteIfExists(checkpointFileName);
+            Files.deleteIfExists(translogFileName);
```

Comment:
You'd have to handle the exceptions (maybe refactor to a helper method) but you could do this in a terminal `.forEach` operation on the stream and avoid collecting to a list.

Reply:
Yeah, this is mostly due to exception handling part. Let me abstract it out to a method as you suggested.